crates/pet-reporter/src/cache.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::environment::get_environment_key;
+use pet_core::{manager::EnvManager, python_environment::PythonEnvironment, reporter::Reporter};
+use std::{
+    collections::HashMap,
+    path::PathBuf,
+    sync::{Arc, Mutex},
+};
+
+pub struct CacheReporter {
+    reporter: Arc<dyn Reporter>,
+    reported_managers: Arc<Mutex<HashMap<PathBuf, EnvManager>>>,
+    reported_environments: Arc<Mutex<HashMap<PathBuf, PythonEnvironment>>>,
```

Comment:
I had removed this from the other reporters, it was duplicated,
hence moved this back into the caching layer, forgot to bring that back.

Reply:
No caching done here, just removal of duplicates